modules/ROOT/pages/dataweave-cookbook-decode-encode.adoc

```diff
@@ -0,0 +1,160 @@
+= Decode and Encode Base64
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: studio, anypoint, transform, transformer, format, xml, metadata, dataweave, data weave, datamapper, dwl, dfl, dw, output structure, input structure, map, mapping
+
+The DataWeave examples show how to convert a file stream into Base64 and to convert a Base64 string into a file stream. They use a PDF file as the input and output.
+//LINK TO DW 1.0 EXAMPLES:
+include::partial$dataweave1-links.adoc[tag=dataweave1Examples]
+
+[[ex01]]
+== Encode a PDF File to Base64
+
+This example performs Base64 encoding on a PDF file. It simulates the transformation in the Read operation from the <<xml_config, XML configuration>> by converting a Base64 file stream (`application/octet-stream`) to Binary and outputting the Binary in the `text/plain` format. 
+
+The example uses the following functions:
+
+* `fromBase64` from the `dw::core::Binaries` module to transform a Base64 string to a binary value.
+*  `readUrl` from the `Core (dw::Core)` module to input a PDF file.
++
+The example assumes that the PDF file is in the `src/main/resources` directory of a Mule project in Anypoint Studio. It provides the name of the PDF file and specifies the MIME type as `"application/octet-stream"`. Using `"binary"` as the second `readUrl` argument is also valid for this example.
+
+//NOTE that we can't yet automate this example. Outputs text format.
+//Need to check if PDF input is supported by validator.
+.DataWeave Script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+import * from dw::core::Binaries
+var myPDF = readUrl("classpath://pdf-test.pdf", "application/octet-stream")
+output text/plain
+---
+toBase64(myPDF as Binary)
+----
+
+.Output (partial results):
+[source,json,linenums]
+----
+JVBERi0xLjYNJeLjz9MNCjM3IDAgb2JqIDw8L0xpbmVhcml6ZWQgMS9MIDIwNTk3L08gNDAvRSAx
+...
+----
+
+
+[[ex02]]
+== Decode Base64
+
+This example simulates the transformation in the Write operation from the <<xml_config, XML configuration>> by converting a Base64 file stream (`application/octet-stream`) to Binary and outputting the Binary in the `application/PDF with binary` format. The example does not generate a PDF file; a Write operation is required to generate such a file. 
+
+The script reads the input PDF as an octet-stream MIME type. From within the call to `fromBase64`, the script coerces that input to a string value before coercing the string to Base64.
+
+The examples use the following functions:
+
+* `fromBase64` in the `dw::core::Binaries` module to transform a Base64 string to a binary value.
+* `readUrl` to input a PDF file.
++
+The example assumes that the PDF file is in the `src/main/resources` directory of a Mule project in Anypoint Studio. It provides the name of the PDF file and specifies the MIME type as `application/octet-stream`. Using the MIME type `"binary"` as the second `readUrl` argument is also valid for this example.
```

Comment:
```suggestion
The examples use the following functions:

* `fromBase64` in the `dw::core::Binaries` module to transform a Base64 string to a binary value.
* `readUrl` to input a PDF file.
+
The script reads the input PDF as an octet-stream MIME type. From within the call to `fromBase64`, the script coerces that input to a string value before coercing the string to Base64.
The example assumes that the PDF file is in the `src/main/resources` directory of a Mule project in Anypoint Studio. It provides the name of the PDF file and specifies the MIME type as `application/octet-stream`. Using the MIME type `"binary"` as the second `readUrl` argument is also valid for this example.
```

Reply:
I would move the sentence afterwards the introduction of the used functions, because the sentence mentions specifically the `fromBase64` function.

Reply:
done, removed duplicative language